lib/uportregistry.js

```diff
@@ -30,31 +28,89 @@ function wrapLowLevelAPI (provider) {
   }
 }
 
-function setIpfsProvider(ipfsProv) {
+function configureIpfs(ipfsProv) {
   if (typeof ipfsProv === 'object') {
     if (typeof ipfsProv.addJSON === 'function' && typeof ipfsProv.catJSON === 'function' ) {
-      ipfs = ipfsProv;
-      return;    
+      return ipfsProv;
     } else if (typeof ipfsProv.add === 'function' && typeof ipfsProv.cat === 'function' ) {
-      ipfs = wrapLowLevelAPI(ipfsProv);
-      return;
+      return wrapLowLevelAPI(ipfsProv);
     } else if (ipfsProv.host) {
-      ipfs = new IPFS(ipfs);
-      return;
+      return new IPFS(ipfs);
     }
   }
   throw new Error('IPFS provider must be either an ipfs-api compliant provider or a configuration hash')
 };
 
-// Default to infura ipfs
-setIpfsProvider(new IPFS({ host: 'ipfs.infura.io', port: 5001, protocol: 'https' }));
+class UportRegistry {
+
+  /**
+  *  Class constructor.
+  *  Creates a new UportRegistry object. The registryAddress is an optional argument and if not specified will be at the moment set to the default ropsten network uport-registry.
+  *
+  *  @memberof        UportRegistry#
+  *  @method          constructor
+  *  @param           {Object}         settings                                                            optional settings containing web3, ipfs and registry settings
+  *  @return          {Object}         self
+  */
+  constructor (settings = {}) {
+    this.ipfs = configureIpfs(settings.ipfs || new IPFS({ host: 'ipfs.infura.io', port: 5001, protocol: 'https' }))
+    RegistryContract.setProvider(settings.web3Prov || new Web3.providers.HttpProvider('https://ropsten.infura.io/uport-persona-lib'))
```

Comment:
@christianlundkvist for this yes, you will have to configure it yourself if you want to use eth-signer or lightwallet